text/0118-search-api.md

```diff
@@ -108,156 +111,273 @@ It returns the number of documents matching the current search query for each gi
-| query                   | String                       | True     |
+| Field                                                 | Type                         | Required |
+|-------------------------------------------------------|------------------------------|----------|
+| [`hits`](#321-hits)                                   | Array[Hit]                   | True     |
+| [`limit`](#322-limit)                                 | Integer                      | True     |
+| [`offset`](#323-offset)                               | Integer                      | True     |
+| [`nbHits`](#324-nbhits)                               | Integer                      | True     |
+| [`exhaustiveNbHits`](#325-exhaustivenbhits)           | Boolean                      | True     |
+| [`facetsDistribution`](#326-facetsdistribution)       | Object                       | False    |
+| [`exhaustiveFacetsCount`](#327-exhaustivefacetscount) | Boolean                      | False    |
+| [`processingTimeMs`](#328-processingtimems)           | Integer                      | True     |
+| [`query`](#329-query)                                 | String                       | True     |
 
-##### 1.2.2.1 `hits`
+#### 3.2.1. `hits`
 
 - Type: Array[Hit]
 - Required: True
 
-Results of the query as an array of documents.
+Results of the search query as an array of documents.
+
+> Hit object represents a matched document as a search result.
+
+> The search parameters `attributesToRetrieve` influence the returned payload for a hit. See [3.1.7. `attributesToRetrieve`](#317-attributestoretrieve) section.
 
-> The search parameters `attributesToRetrieve` influence the returned payload for a document as a search result. See 1.2.1.7 `attributesToRetrieve` section.
+A search result can host special properties. See [3.2.1.1. `hit` Special Properties](#3211-hits-special-properties) section.
 
-> A Hit object that represents a document within the search results can host special attributes. See 1.2.2.9 `hits` special fields section.
+##### 3.2.1.1. `hit` Special Properties
 
-##### 1.2.2.2 `limit`
+| Field                                | Type        | Required |
+|--------------------------------------|-------------|----------|
+| [`_geoDistance`](#32111-geodistance) | Integer     | False    |
+| [`_formatted`](#32112-formatted)     | Object      | False    |
+| [`_matchesInfo`](#32113-matchesinfo) | Object      | False    |
+
+###### 3.2.1.1.1. `_geoDistance`
+
+- Type: Integer
+- Required: False
+
+Search queries using `_geoPoint` returns a `_geoDistance` field containing the distance in meters between the document `_geo` coordinates and the specified `_geoPoint`.
+
+> See [GeoSearch](0059-geo-search.md)
+
+###### 3.2.1.1.2. `_formatted`
+
+- Type: Object
+- Required: False
+
+`_formatted` returns highlighted and cropped attributes specified in `attributesToHighlight` and/or `attributesToCrop` of a search result.
```

Comment:
If I understand correctly, only the `attributesToHighlight` and `attributesToCrop` are put in `_formatted`. If `*` is set as `attributesToHighlight` or `attributesToCrop`, _formatted will return `attributesToRetrieve`?

Reply:
From what I remember, attributesToRetrieves only impacts the fields at the root of the hit. AttributesToHighlight and attributesToCrop applies only in `_formatted`.

Reply:
The behavior is unchanged from today. I have added some important instructions about this (extracted from the deleted _formatted-field-behavior specification).

- If `attributesToHighlight` and `attributesToCrop` are not set, `_formatted` is not returned.
- If cumulated fields in `attributesToHighlight` and `attributesToCrop` resolve to only having non-existent fields, `_formatted` is not returned.
- If `attributesToRetrieve` is equal to `*` and `attributesToHighlight` or `attributesToCrop` are equals to `*`, `_formatted` is returned and contains `attributesToRetrieve` fields then compute highlights and crops on each received fields.
- If `attributesToRetrieve` is equal to `*` and `attributesToHighlight` or `attributesToCrop` contains a set of fields, `_formatted` is returned and contains `attributesToRetrieve` fields but only compute highlights and crops on fields declared in `attributesToHighlight` or `attributesToCrop`.

Reply:
Yes!